chromium_src/chrome/browser/ui/managed_ui.cc

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/enterprise/util/managed_browser_utils.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/policy/core/common/policy_map.h"
+#include "components/policy/core/common/policy_namespace.h"
+#include "components/policy/core/common/policy_service.h"
+#include "components/policy/core/common/policy_types.h"
+
+#define ShouldDisplayManagedUi ShouldDisplayManagedUi_ChromiumImpl
+#include <chrome/browser/ui/managed_ui.cc>
+#undef ShouldDisplayManagedUi
+
+namespace brave_policy {
+
+bool ShouldHideManagedUI(const policy::PolicyMap& policies) {
+  // Empty policy map means no management at all
+  if (policies.empty()) {
+    return false;
+  }
+
+  // Check all policies, not just BraveOrigin-related ones
+  for (const auto& [policy_name, entry] : policies) {
+    if (entry.source != policy::POLICY_SOURCE_BRAVE) {
+      return false;  // Found non-Brave policy
+    }
+  }
+  return true;
+}
+
+}  // namespace brave_policy
+
+// Our override implementation
+bool ShouldDisplayManagedUi(Profile* profile) {
+  // Check if we should hide due to Brave-only management
+  if (enterprise_util::IsBrowserManaged(profile)) {
```

Comment:
nit: what if we call the original function and perform an early return instead of calling this function?

```
if (!ShouldDisplayManagedUi_ChromiumImpl(profile)) {
  return false;
}
```

Reply:
Sounds good. Updated in the latest commit.